torch/_tensor.py

```diff
@@ -560,6 +560,18 @@ def __floordiv__(self, other):
     def __rfloordiv__(self, other):
         return torch.floor_divide(other, self)
 
+    @_wrap_type_error_to_not_implemented
+    def __rlshift__(self, other):
+        if has_torch_function_variadic(self, other):
```

Comment:
here (and in all other `__r*__` functions) you should not handle torch function twice, in the wrapper and in the function itself. Moreover, the wrapper is using the slowest variant (`has_torch_function`), you can create a faster wrapper that uses `has_torch_function_variadic`

Reply:
Fixed in e2df19d69201f475e3ff948528c69548b1c566bf. Thanks!